src/relay/op/memory/device_copy.cc

```diff
@@ -103,6 +107,19 @@ DeviceCopyProps GetDeviceCopyProps(const CallNode* call_node) {
     } else {
       return {call_node->args[0], src_dev_type, dst_dev_type};
     }
+  } else if (call_node->op == CallLoweredOp()) {
+    /* Get device props for a TIR function */
+    CallLoweredProps call_lowered_props = GetCallLoweredProps(call_node);
+
+    if (call_lowered_props.attrs.metadata.count("source_device") == 1 &&
```

Comment:
confirming looks like we can avoid this here
(eventually GetPrimitiveDeviceCopyProps from device_domains.cc will need to get moved alongside this one (and renamed) but no need here)